### ISSUE TITLE
Glib icon, caught exception

### ISSUE THREAD

**Issue by Zeckmathederg** [SkyTemple/skytemple#646]
On my linux from scratch system. The last install of it seemed to install skytemple just fine, but this one, using the recommended pip install ... command in the README.rst succeeds but results in most of the icons except the refresh button, the 3 bar icon, crying duskako icon, and the banner about updates, are missing. Launching from the terminal gives the gtk warning:
```
(skytemple:9438): Gtk-WARNING **: 00:36:58.716: Could not find the icon 'skytemple-debugger-debug-symbolic-ltr'. The 'hicolor' theme
was not found either, perhaps you need to install it.
You can get a copy from:
	http://icon-theme.freedesktop.org/releases
```
And attempting to open a rom will reveal an uncaught exception, I will post what i see from the html file in my /tmp dir here:
```



Error | Python 3.11.2: /usr/bin/python3.11Sun Oct 15 00:45:23 2023
-- | --


<p><strong>Error</strong>: gtk-icon-theme-error-quark: Icon 'skytemple-image-loading-symbolic' not present in theme Adwaita (0)
<br><tt><small>     </small> </tt>add_note =
&lt;built-in method add_note of Error object&gt;
<br><tt><small>     </small> </tt>args =
("Icon 'skytemple-image-loading-symbolic' not present in theme Adwaita",)
<br><tt><small>     </small> </tt>code =
0
<br><tt><small>     </small> </tt>copy =
&lt;bound method GError.copy of GLib.Error('Icon 's...theme Adwaita', 'gtk-icon-theme-error-quark', 0)&gt;
<br><tt><small>     </small> </tt>domain =
'gtk-icon-theme-error-quark'
<br><tt><small>     </small> </tt>matches =
&lt;bound method gerror_matches of GLib.Error('Icon...theme Adwaita', 'gtk-icon-theme-error-quark', 0)&gt;
<br><tt><small>     </small> </tt>message =
"Icon 'skytemple-image-loading-symbolic' not present in theme Adwaita"
<br><tt><small>     </small> </tt>new_literal =
&lt;function gerror_new_literal&gt;
<br><tt><small>     </small> </tt>with_traceback =
&lt;built-in method with_traceback of Error object&gt;



</p>Error 	Python 3.11.2: /usr/bin/python3.11
Sun Oct 15 00:45:23 2023

A problem occurred in a Python script. Here is the sequence of function calls leading up to the error, in the order they occurred.
 [/home/zeckma/.local/lib/python3.11/site-packages/skytemple/core/rom_project.py](file:///home/zeckma/.local/lib/python3.11/site-packages/skytemple/core/rom_project.py) in <lambda>()
    118             await cls._current.load()
    119             if main_controller:
=>  120                 GLib.idle_add(lambda: main_controller.on_file_opened())
    121         except BaseException as ex:
    122             exc_info = sys.exc_info()
global GLib = <GLibProxyModule <IntrospectionModule 'GLib' from '/usr/lib/girepository-1.0/GLib-2.0.typelib'>>, GLib.idle_add = <function idle_add>, main_controller = <skytemple.controller.main.MainController object>, main_controller.on_file_opened = <bound method MainController.on_file_opened of <skytemple.controller.main.MainController object>>
 [/home/zeckma/.local/lib/python3.11/site-packages/skytemple/controller/main.py](file:///home/zeckma/.local/lib/python3.11/site-packages/skytemple/controller/main.py) in on_file_opened(self=<skytemple.controller.main.MainController object>)
    318         project = RomProject.get_current()
    319         assert project is not None
=>  320         project.get_sprite_provider().init_loader(self._window.get_screen())
    321 
    322         self._init_window_after_rom_load(os.path.basename(project.filename))
project = <skytemple.core.rom_project.RomProject object>, project.get_sprite_provider = <bound method RomProject.get_sprite_provider of <skytemple.core.rom_project.RomProject object>>, ).init_loader undefined, self = <skytemple.controller.main.MainController object>, self._window = <Gtk.Window object at 0x7f032de11f40 (GtkWindow at 0x560bb5d9ca50)>, self._window.get_screen = gi.FunctionInfo(get_screen)
 [/home/zeckma/.local/lib/python3.11/site-packages/skytemple/core/sprite_provider.py](file:///home/zeckma/.local/lib/python3.11/site-packages/skytemple/core/sprite_provider.py) in init_loader(self=<skytemple.core.sprite_provider.SpriteProvider object>, screen=<GdkX11.X11Screen object at 0x7f030ec6ca80 (GdkX11Screen at 0x560bb514c060)>)
    188         icon_theme: Gtk.IconTheme = Gtk.IconTheme.get_for_screen(screen)
    189         # Loader icon
=>  190         loader_icon: GdkPixbuf.Pixbuf = icon_theme.load_icon(
=>  191             'skytemple-image-loading-symbolic', 24, Gtk.IconLookupFlags.FORCE_SIZE
=>  192         ).copy()
loader_icon undefined, global GdkPixbuf = <GdkPixbufProxyModule <IntrospectionModule 'GdkP...usr/lib/girepository-1.0/GdkPixbuf-2.0.typelib'>>, GdkPixbuf.Pixbuf = <class 'gi.overrides.GdkPixbuf.Pixbuf'>, icon_theme = <Gtk.IconTheme object at 0x7f031e5a74c0 (GtkIconTheme at 0x560bb52cc5a0)>, icon_theme.load_icon = gi.FunctionInfo(load_icon), global Gtk = <GtkProxyModule <IntrospectionModule 'Gtk' from '/usr/lib/girepository-1.0/Gtk-3.0.typelib'>>, Gtk.IconLookupFlags = <class 'gi.repository.Gtk.IconLookupFlags'>, Gtk.IconLookupFlags.FORCE_SIZE = <flags GTK_ICON_LOOKUP_FORCE_SIZE of type Gtk.IconLookupFlags>, ).copy undefined

Error: gtk-icon-theme-error-quark: Icon 'skytemple-image-loading-symbolic' not present in theme Adwaita (0)
      add_note = <built-in method add_note of Error object>
      args = ("Icon 'skytemple-image-loading-symbolic' not present in theme Adwaita",)
      code = 0
      copy = <bound method GError.copy of GLib.Error('Icon 's...theme Adwaita', 'gtk-icon-theme-error-quark', 0)>
      domain = 'gtk-icon-theme-error-quark'
      matches = <bound method gerror_matches of GLib.Error('Icon...theme Adwaita', 'gtk-icon-theme-error-quark', 0)>
      message = "Icon 'skytemple-image-loading-symbolic' not present in theme Adwaita"
      new_literal = <function gerror_new_literal>
      with_traceback = <built-in method with_traceback of Error object>
```
Could this be a faulty install of skytemple and would I be missing an installation or library of any sort?

**Comment by theCapypara** [comment]
Hm, there seems to be something wrong with icon themes on your machine. It can't find hicolor and probably also no other theme. It shouldn't crash when it can't find the SkyTemple icons either, but it probably can't even find a fallback icon.

**Comment by Zeckmathederg** [comment]
Does it search in the usr directory ie. `/usr/share/icons`?

**Comment by Zeckmathederg** [comment]
I have the intelligence of a rock that sunk 2k kilometers under water level. I apparently had never done this, and looking in my directory tree, it seemed like everything was fine because /usr/share/icons/hicolor had contents and so did ~/.local/share/icons/hicolor, but I needed to install hicolor-icon-theme. In other words yeah I just didn't install a package. After I did, SkyTemple worked fine. Thanks for pushing me in the right direction!
